Control/DeepSeq.hs

```diff
@@ -299,10 +320,35 @@ class NFData a where
     rnf :: a -> ()
 
 #if __GLASGOW_HASKELL__ >= 702
-    default rnf :: (Generic a, GNFData (Rep a)) => a -> ()
-    rnf = grnf . from
+    default rnf :: (Generic a, GNFData Zero (Rep a)) => a -> ()
+    rnf = grnf RnfArgs0 . from
+#endif
+
+
+-- | A class of functors that can be fully evaluated.
+--
+-- @since 1.4.3.0
+class NFData1 f where
```

Comment:
It'd be nice to have some documentation describing how to derive `NFData1` generically, similarly to the current documentation that `NFData` has [here](https://github.com/haskell/deepseq/blob/d496ce4ca1a1141dda3fe90b129d842e7c52c7e4/Control/DeepSeq.hs#L240).